### ISSUE TITLE
Clarification on Input Requirements for AIDO.Cell-100M

### ISSUE THREAD

**Issue by lkrishnan-czi** [genbio-ai/ModelGenerator#5]
Hi,

I’m currently testing the AIDO.Cell-100M model to better understand its usage and would like to confirm a few implementation details.
 Below is a snippet of the code I’m working with to generate cell embeddings from the model.

```
from modelgenerator.tasks import Embed
model = Embed.from_config({"model.backbone": "aido_cell_100m"}).eval()
model = model.backbone

batch_size = 1
seq_len = 19264

# Example input tensors
input_ids = torch.randint(0, 1000, (batch_size, seq_len)).to('cuda') 
attention_mask = torch.ones((batch_size, seq_len)).to('cuda') 

# Forward pass through the Embed wrapper with mixed precision
with autocast():
    embeddings = model.forward(input_ids, attention_mask) # shape bs x 19264 x 640
```

I have a couple of questions about the input data:
- Can input_ids represent raw gene counts directly, or is pre-processing (e.g., normalization) required?
- For gene mapping, I’m using the file [OS_scRNA_gene_index.19264.tsv](https://github.com/biomap-research/scFoundation/blob/main/OS_scRNA_gene_index.19264.tsv). Is this the correct vocabulary file for constructing input_ids, and does each index in the sequence correspond to a fixed gene position based on the order in this file?
- How do you handle instances when some gene_ids are missing in an evaluation dataset?

Thank you for your help in clarifying these points.
Let me know if more information is needed!

**Comment by cnellington** [comment]
Hi @lkrishnan-czi, I would strongly recommend using the CLI interface if you're using built-in tasks like Embed. I can point you to a few tutorials and code snippets to clarify.

One of our design decisions in ModelGenerator was to treat tokenization as a property of the backbone, so tokenization has to happen at runtime once a backbone is chosen. All tasks will pass the batch through a transform() function that applies the backbone's tokenizer before calling forward(). See the predict_step in the TaskInterface.
https://github.com/genbio-ai/ModelGenerator/blob/5a7764ec9fcc419e97713de37a98631fe3b137d6/modelgenerator/tasks/base.py#L224-L238

So tasks should always be used through the CLI, or at least the pytorch lightning Trainer object. I recommend the CLI because one of the experiments from AIDO.Cell that we open-sourced with this repo can be quickly adapted for embedding by swapping out the task being used. See the experiment walkthrough + config here
https://github.com/genbio-ai/ModelGenerator/tree/main/experiments/AIDO.Cell

You should be able to adapt this to embed instead of train a classifier by just doing
```
mgen predict --config experiments/AIDO.Cell/cell_type_classification.yaml --model Embed
```

We also have examples of using backbones directly for embedding with HuggingFace, although unfortunately not for AIDO.Cell. See the bottom of this page for examples
https://huggingface.co/genbio-ai/AIDO.DNA-7B

Hope this helps.

**Comment by lkrishnan-czi** [comment]
Thank you for your response!

For the aido-cell-100m model, it seems that the tokenizer is essentially a dummy function, as the model directly consumes expression values:
[Backbone implementation](https://github.com/genbio-ai/ModelGenerator/blob/main/modelgenerator/backbones/backbones.py#L691-L711).

Regarding the Embed task, I don’t see any additional processing applied beyond the raw counts:
[Embed task implementation](https://github.com/genbio-ai/ModelGenerator/blob/main/modelgenerator/tasks/tasks.py#L1583-L1605).

Additionally, the CLI instructions don’t clearly explain how to handle custom adata files. The [YAML config](https://github.com/genbio-ai/ModelGenerator/blob/main/experiments/AIDO.Cell/cell_type_classification.yaml) appears to define paths for fine-tuning on a cell type classification task - do we need all the dataset splits? and is this meant to also work out of the box for embedding task?

Would it be possible to provide a straightforward example of how to embed an arbitrary adata file using the aido-cell-100m model? That would be extremely helpful—thank you!
